src/spaceObjects/asteroid.cpp

```diff
@@ -80,10 +80,10 @@ void Asteroid::drawOnRadar(sp::RenderTarget& renderer, glm::vec2 position, float
     if (size != getRadius())
         setRadius(size);
 
-    float size = getRadius() * scale / 64.0f;
-    if (size < 0.2f)
-        size = 0.2f;
-    renderer.drawSprite("radar/blip.png", position, size * 32.0f, glm::u8vec4(255, 200, 100, 255));
+    renderer.drawSprite("radar/blip.png", position, std::max(6.0f, (getRadius() * 2.0f) * scale), glm::u8vec4(255, 200, 100, 255));
```

Comment:
I would expect a min as well, or a std::clamp in that case. As it's nice to always be able to see the asteroids, even when zoomed out.

The ECS branch "generalized" this with a RadarTrace component that has a min/max/scaled size properties, and there I already applied this for asteroids.

Reply:
> I would expect a min as well, or a std::clamp in that case. As it's nice to always be able to see the asteroids, even when zoomed out.

`std::max` returns the larger of the two given values, so when the scale is low (zoomed out) or the asteroid is small, the sprite is always drawn at least 6 pixels wide.

This is visible in the screenshots; the 10-radius asteroid at 0.09 scale (max GM screen zoom) would be ~2 pixels, but is rendered 6 pixels wide, or at about the same size as the master branch.

I use 6 pixels because this is close to the result of the existing `if size < 0.2 then size = 0.2` gate; 0.2 * 32 = 6.4.

Science "Zoom 1.0x", labeled by asteroid radius:

![image](https://user-images.githubusercontent.com/19192104/221026310-c38d6974-0ba8-426e-952a-464c339dc353.png)

"Zoom 6.0x": (scale ~0.09)

![image](https://user-images.githubusercontent.com/19192104/221026354-5052fd07-a472-4585-99db-0125a1173996.png)

Relay "Zoom 1.0x": (scale ~0.01)

![image](https://user-images.githubusercontent.com/19192104/221026431-8b1dadce-1dd7-4211-9e77-38313744b6cc.png)

```
[INFO    ]: std::max(6.0f, (getRadius() * 2.0f) * scale): 6.00
getRadius(): 10.00
scale: 0.01
[INFO    ]: std::max(6.0f, (getRadius() * 2.0f) * scale): 6.00
getRadius(): 121.19
scale: 0.01
[INFO    ]: std::max(6.0f, (getRadius() * 2.0f) * scale): 6.00
getRadius(): 253.12
scale: 0.01
[INFO    ]: std::max(6.0f, (getRadius() * 2.0f) * scale): 9.00
getRadius(): 500.00
scale: 0.01
```

Adding `min` or using `clamp` would allow setting a limit on the maximum size of the asteroid sprite, but the radar sprite not accurately reflecting large asteroids' collidable size. I'm not sure what the purpose of an upper limit would be.

Reply:
ah, my mistake, I was thinking about it wrong :)